src/algorithms/tracking/TrackParamTruthInit.cc

```diff
@@ -92,7 +93,7 @@ eicrecon::TrackParamTruthInit::produce(const edm4hep::MCParticleCollection* mcpa
         // Insert into edm4eic::TrackParameters, which uses numerical values in its specified units
         auto track_parameter = track_parameters->create();
         track_parameter.setType(-1); // type --> seed(-1)
-        track_parameter.setLoc({0.0, 0.0}); // 2d location on surface [mm]
+        track_parameter.setLoc({static_cast<float>(std::hypot(v.x, v.y)), static_cast<float>(v.z)}); // 2d location on surface [mm]
```

Comment:
So coordinates in
https://github.com/eic/EICrecon/blob/3a54c0125862d80326f1b6b5b273b793acbbe3b6/src/global/tracking/CKFTracking_factory.cc#L83
can be effectively provided with
https://github.com/eic/EICrecon/blob/3a54c0125862d80326f1b6b5b273b793acbbe3b6/src/global/tracking/CKFTracking_factory.cc#L65-L66
?

Reply:
Not quite. The non-zero three vector position with zero two-vector on the surface specifies a unique point. The zero three vector with non-zero two-vector on the surface specifies a circle (of closest approach).